api/src/services/critterbase-service.ts

```diff
@@ -0,0 +1,249 @@
+  ['taxon-collection-categories']: `${xref}/taxon-collection-categories`,
+  ['taxon-marking-body-locations']: `${xref}/taxon-marking-body-locations`
+} as const;
+
+export type ICbRouteKey = keyof typeof CbRoutes;
+
+export const CRITTERBASE_API_HOST = process.env.CB_API_HOST || ``;
+const CRITTER_ENDPOINT = '/critters/';
+const BULK_ENDPOINT = '/bulk';
+const SIGNUP_ENDPOINT = '/signup';
+const FAMILY_ENDPOINT = '/family';
+
+export class CritterbaseService {
+  user: ICritterbaseUser;
+  keycloak: KeycloakService;
+
+  constructor(user: ICritterbaseUser) {
+    this.user = user;
+    this.keycloak = new KeycloakService();
+  }
+
+  async getToken(): Promise<string> {
+    const token = await this.keycloak.getKeycloakServiceToken();
+    return token;
+  }
+
+  getUserHeader(): { user: string } {
+    return { user: JSON.stringify(this.user) };
+  }
+
+  async makeGetRequest(endpoint: string, params: QueryParam[]) {
+    let url = `${CRITTERBASE_API_HOST}${endpoint}`;
+    for (let i = 0; i < params.length; i++) {
+      url += (i === 0 ? '?' : '&') + params[i].key + '=' + params[i].value;
+    }
+    console.log('Actual URL we will hit ' + url);
+    const token = await this.getToken();
+    const response = await axios.get(url, {
+      headers: {
+        authorization: `Bearer ${token}`,
+        ...this.getUserHeader()
+      }
+    });
+    return response.data;
+  }
+
+  async makePostPatchRequest(method: 'post' | 'patch', endpoint: string, data?: any) {
+    const url = `${CRITTERBASE_API_HOST}${endpoint}`;
+    const token = await this.getToken();
+    const response = await axios[method](url, data, {
+      headers: {
+        authorization: `Bearer ${token}`,
+        ...this.getUserHeader()
+      }
+    });
+    return response.data;
+  }
+
+  async getLookupValues(route: ICbRouteKey, params: QueryParam[]) {
+    return this.makeGetRequest(`${CbRoutes[route]}`, params);
+  }
+
+  async getTaxonMeasurements(taxon_id: string) {
+    return this.makeGetRequest(`${CbRoutes['taxon-measurements']}`, [{ key: 'taxon_id', value: taxon_id }]);
+  }
+
+  async getTaxonBodyLocations(taxon_id: string) {
+    return this.makeGetRequest(`${CbRoutes['taxon-marking-body-locations']}`, [
+      { key: 'taxon_id', value: taxon_id },
+      { key: 'format', value: 'asSelect' }
+    ]);
+  }
+
+  async getQualitativeOptions(taxon_measurement_id: string, format = 'asSelect') {
+    return this.makeGetRequest(`${CbRoutes['taxon-qualitative-measurement-options']}`, [
+      { key: 'taxon_measurement_id', value: taxon_measurement_id },
+      { key: 'format', value: format }
+    ]);
+  }
+
+  async getFamilies() {
+    return this.makeGetRequest(FAMILY_ENDPOINT, []);
+  }
+
+  async getFamilyById(family_id: string) {
+    return this.makeGetRequest(`${FAMILY_ENDPOINT}/${family_id}`, []);
+  }
+
+  async getCritter(critter_id: string) {
+    return this.makeGetRequest(CRITTER_ENDPOINT + critter_id, [{ key: 'format', value: 'detail' }]);
```

Comment:
Nit: Avoid using `+` operator to join strings and use `${CRITTER_ENDPOINT} ${critter_id}`